### ISSUE TITLE
Add more markers

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#1500]
Having implemented permanent markers, having more types of markers would be nice:

- Arrows
- Ellipses
- Scalebar: could use the matplotlib sizebar http://matplotlib.org/devdocs/mpl_toolkits/axes_grid1/api/anchored_artists_api.html#mpl_toolkits.axes_grid1.anchored_artists.AnchoredSizeBar
- Patches (http://matplotlib.org/examples/api/patch_collection.html)
- Collections: for example a PointCollection, which would avoid clutter and improve plotting performance when adding a large amount of points

**Comment by sem-geologist** [comment]
again reinventing the wheel... all those things are present in pyqtgraph...

**Comment by francisco-dlp** [comment]
Yes, but currently we're using matplotlib. They're obviously also implemented in matplotlib, but HyperSpy ones are multidimensional and can be stored in our hdf5 specification, so I am not sure if this is about reinventing the wheel...
